### PR TITLE
feat: add Windows screen record

### PR DESCRIPTION
**Pull request by KazuCocoa**

### CODE REVIEW

File: lib/appium_lib_core/windows/device.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        # rubocop:disable Metrics/LineLength
+
+        # @since Appium 1.18.0
+        # @!method start_recording_screen(remote_path: nil, user: nil, pass: nil, method: 'PUT', force_restart: nil, time_limit: nil, fps: nil, preset: nil, video_filter: nil, capture_clicks: nil, capture_cursor: nil, audio_input: nil)
+        #
+        # Record the display in background while the automated test is running.
+        # This method requires FFMPEG (https://www.ffmpeg.org/download.html) to be installed and present in PATH.
+        # The resulting video uses H264 codec and is ready to be played by media players built-in into web browsers.
+        #
+        # @param [String] remote_path The path to the remote location, where the resulting video should be uploaded.
```
Comment (mykola-mokhnach) [inline lib/appium_lib_core/windows/device.rb]: actually the method will just return silently if there is an active screen record running.

File: lib/appium_lib_core/windows/device.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        #                             fit into the available process memory.
+        #                             This option only has an effect if there is screen recording process in progreess
+        #                             and +forceRestart+ parameter is not set to +true+.
+        # @param [String] user The name of the user for the remote authentication.
+        # @param [String] pass The password for the remote authentication.
+        # @param [String] method The http multipart upload method name. The 'PUT' one is used by default.
+        # @param [Boolean] force_restart Whether to try to catch and upload/return the currently running screen recording
+        #                                (+false+, the default setting on server) or ignore the result of it
+        #                                and start a new recording immediately (+true+).
+        # @param [String] time_limit Recording time. 180 seconds is by default.
```
Comment (mykola-mokhnach) [inline lib/appium_lib_core/windows/device.rb]: for windows it's 600 seconds

**Comment by mykola-mokhnach** [comment]
Sorry for the miscommunication @KazuCocoa 
In https://github.com/appium/java-client/pull/1313/commits/d3a0c4a2d0f77aca8c64b1823b13b4af0b3b31a6 I've made the UploadOptions hidden for startRecord in Windows. I think the fact it returns a video recording is a side effect and should be avoided, so in Windows this method simply ignores the currently running video recording and returns silently or starts a new one if the previous recording has been already finished. If forced restart is enabled then the previous recording is stopped forcefully and a new one is started, although nothing is returned in such case as well.

**Comment by mykola-mokhnach** [comment]
https://github.com/appium/appium-windows-driver/blob/e3269bfffa159e0acd68b2711415985ee3527863/lib/commands/record-screen.js#L260 is the current Windows logic

**Comment by KazuCocoa** [comment]
ah, okay 👌

**Pull request merged**